### PR TITLE
Encoder calibration + universal build

### PR DESCRIPTION
**Pull request by Bernstern**
# Draft

Added a simple average calibration for the absolute encoders that you can bind to a button or we could throw on the smart dashboard. 

Also fixed the symlink :)

Possible todo is refactoring this into a command.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,6 +1,6 @@
 plugins {
     id "java"
-    id "edu.wpi.first.GradleRIO" version "2024.1.1"
+    id "edu.wpi.first.GradleRIO" version "2024.2.1"
```
Comment (Mineinjava) [inline build.gradle:3]: Good catch

File: src/main/java/frc/robot/Robot.java [hunk 2]
```diff
@@ -5,58 +5,209 @@
...
+      this.gyro.reset();
+    }
+
+		double[] pos = NetworkTableInstance.getDefault().getTable("limelight").getEntry("botpose").getDoubleArray(new double[6]);
+		SmartDashboard.putNumberArray("Limelight Pos", pos);
+		SmartDashboard.putNumber("LX", -pos[1] * Constants.METERS_TO_INCHES);
+		SmartDashboard.putNumber("LY", pos[0] * Constants.METERS_TO_INCHES);
+
+		if(pos[0] != 0 && pos[1] != 0){
+			odometry.setPose(new Pose2d(-pos[1]* Constants.METERS_TO_INCHES, pos[0]* Constants.METERS_TO_INCHES, -this.gyro.getAngle() * Constants.DEG_TO_RAD));
```
Comment (Mineinjava) [inline src/main/java/frc/robot/Robot.java:168]: So we are moving this here now?

**Review by Mineinjava — COMMENTED** [review]
This seems to undo the declarative refactor?

**Comment by Bernstern** [comment]
We can close this since it looks like we can't do encoder calibration like this + I merged in the universal build change on the refactor branch

**Pull request closed without merging**